tests/test_patroni.py

```diff
@@ -202,3 +203,38 @@ def test_check_psycopg(self):
             self.assertRaises(SystemExit, check_psycopg)
         with patch('builtins.__import__', mock_import):
             self.assertRaises(SystemExit, check_psycopg)
+
+    def test_ensure_unique_name(self):
+        # None/empty cluster implies unique name
+        with patch('patroni.dcs.AbstractDCS.get_cluster', Mock(return_value=None)):
+            self.p.ensure_unique_name()
```

Comment:
I would prefer checking that sys.exit was not called (because when I remove line 240 here, tests still pass, this means we are likely to miss the fact non-erroneous test cases suddenly start to exit with code 1)

Or, even better, per @CyberDem0n suggestion, switch to checking wether an [exception was raised ](https://github.com/zalando/patroni/blob/master/tests/test_patroni.py#L159)/`None` was returned by `ensure_unique_name()`